File: software/examples/vbnet/ExampleCallback.vb
Imports System
Imports Tinkerforge

Module ExampleCallback
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Hall Effect Bricklet

    ' Callback subroutine for edge count callback
    Sub EdgeCountCB(ByVal sender As BrickletHallEffect, ByVal count As Long, _
                    ByVal value As Boolean)
        Console.WriteLine("Count: " + count.ToString())
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim he As New BrickletHallEffect(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register edge count callback to subroutine EdgeCountCB
        AddHandler he.EdgeCountCallback, AddressOf EdgeCountCB

        ' Set period for edge count callback to 0.05s (50ms)
        ' Note: The edge count callback is only called every 0.05 seconds
        '       if the edge count has changed since the last call!
        he.SetEdgeCountCallbackPeriod(50)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/delphi/ExampleCallback.pas
program ExampleCallback;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletHallEffect;

type
  TExample = class
  private
    ipcon: TIPConnection;
    he: TBrickletHallEffect;
  public
    procedure EdgeCountCB(sender: TBrickletHallEffect;
                          const edge_count: longword; const value: Boolean);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change to your UID }

var
  e: TExample;

{ Callback function for edge count callback }
procedure TExample.EdgeCountCB(sender: TBrickletHallEffect;
                               const edge_count: longword; const value: Boolean);
begin
  WriteLn(Format('Edge Count: %d', [edge_count]));
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  he := TBrickletHallEffect.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Set Period for edge_count callback to 0.05s (50ms)
    Note: The edge_count callback is only called every 50ms if the
          edge_count has changed since the last call! }
  he.SetEdgeCountCallbackPeriod(50);

  { Register edge_count callback to procedure EdgeCountCB }
  he.OnEdgeCount := {$ifdef FPC}@{$endif}EdgeCountCB;

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.
